tools/logger/logger.c

```diff
@@ -339,7 +411,15 @@ int main(int argc, char *argv[])
 			goto out;
 		}
 	} else if (config.in_file) {
-		config.in_fd = fopen(config.in_file, "rb");
+		/* In the future we could add an option to force waiting
+		 * for _any_ input file, not just for /sys/kernel/debug/sof/[e]trace
+		 */
+		if (strncmp(config.in_file, SYS_DEBUG, strlen(SYS_DEBUG)))
+			config.in_fd = fopen(config.in_file, "rb");
+		else
+			config.in_fd = wait_fopen(SYS_DEBUG,
+						  config.in_file + strlen(SYS_DEBUG) + 1);
```

Comment:
I would be tempted to move this whole logic down into a custom `logger_fopen()` or something.
That way you would not need to re-construct the config.in_file into a new malloced buffer?

Reply:
Future possible improvement? Keep in mind we want to discard this logger entirely in the longer term.